File: DeSS_T_Frontend/src/app/models/Network.ts
// ------------------- NETWORK MODEL --------------------
export interface NetworkModel {
  network_model_id?: string;
  Network_model?: string; // API field (not stored in DB)
  StationPair?: StationPair[]; // matches Go json:"StationPair"
  Station_detail?: StationDetail[]; // matches Go json:"Station_detail"
}

// ------------------- STATION DETAIL --------------------
export interface StationDetail {
  station_detail_id?: string; // Go json:"station_detail_id"
  StationID?: string; // Go json:"StationID"
  name?: string; // Go json:"name"
  StationName?: string; // Go json:"StationName"
  Location?: GeoPoint; // Go json:"Location"
  lat?: number; // Go json:"lat"
  lon?: number; // Go json:"lon"
  station_id_osm?: string; // Go json:"station_id_osm"
}

// ------------------- STATION PAIR --------------------
export interface StationPair {
  StationPairID: string; // Go json:"StationPairID"
  FstStation: string; // Go json:"FstStation" (station ID, maps to FstStationID in DB)
  SndStation: string; // Go json:"SndStation" (station ID, maps to SndStationID in DB)
  route_between_id: string; // Go json:"route_between_id"
  network_model_id?: string; // Go json:"network_model_id"
  RouteBetween?: RouteBetween; // Go json:"RouteBetween" (includes Distance and TravelTime)
}

// ------------------- ROUTE BETWEEN --------------------
export interface RouteBetween {
  RouteBetweenID: string; // Go json:"RouteBetweenID"
  TravelTime: number; // Go json:"TravelTime"
  Distance: number; // Go json:"Distance"
}

// ------------------- ROUTE PATH --------------------
export interface RoutePath {
  route_path_id: string;
  name: string;
  color: string;
  route_scenario_id: string;
  route: string; // GeoJSON as string
  orders?: Order[];
}

// ------------------- ORDER --------------------
export interface Order {
  order_id: string;
  order: number; // Sequential order number
  station_pair_id: string;
  route_path_id: string;
  route_path?: RoutePath;
  station_pair?: StationPair;
}

// ------------------- Helper Types --------------------
export type LatLng = [number, number];

export interface GeoPoint {
  type: "Point";
  coordinates: [number, number];
}

export interface GeoLineString {
  type: "LineString";
  coordinates: [number, number][];
}

File: DeSS_T_Backend-go/models/network.go
package models

type Network_Model struct {
    NetworkModel  string          `json:"Network_model"`
    StationDetail []Station_Detail `json:"Station_detail"`
    StationPair   []Station_Pair   `json:"StationPair"`
}

/* ─────────────── Station Detail ─────────────── */

type Station_Detail struct {
    StationID   string   `json:"StationID"`
    StationName string   `json:"StationName"`
    Location    GeoPoint `json:"location"`
    Lat         string   `json:"Lat"`
    Lon         string   `json:"Lon"`
}

type GeoPoint struct {
    Type        string     `json:"type"`
    Coordinates [2]float64 `json:"coordinates"`
} 

/* ─────────────── Station Pair ─────────────── */

type Station_Pair struct {
    StationPairID string        `json:"StationPairID"`
    FstStation   string        `json:"FstStation"`
    SndStation   string        `json:"SndStation"`
    RouteBetween Route_Between  `json:"RouteBetween"`
}

type Route_Between struct {
    RouteBetweenID string         `json:"RouteBetweenID"`
    TravelTime     float64        `json:"TravelTime"`
    Route          GeoLineString  `json:"Route"`
    Distance       float64        `json:"Distance"`
}

type GeoLineString struct {
    Type        string        `json:"type"`
    Coordinates [][2]float64  `json:"coordinates"`
}

